app/hooks/useQuestionStateInUrl.ts

```diff
@@ -158,7 +160,15 @@ export default function useQuestionStateInUrl(minLogo: boolean, initialQuestions
       if (stateString == newState) return
       const newSearchParams = new URLSearchParams(remixSearchParams)
       newSearchParams.set('state', newState)
-      history.replaceState(newState, '', '?' + newSearchParams.toString())
+
+      const newUrl = '?' + newSearchParams.toString()
+      if (allowBrowserBackToInitialStateRef.current) {
```

Comment:
what is this for?

Reply:
(will answer in next comment)

Reply:
sneaking in a fix for situation when someone starts at homepage or some other page from direct link, then opens a bunch of stuff and clicks browser back button hoping to land on the initial page ... but currently, we only replace state so back button takes you to previous domain and then forward takes you to final state not initial state, page refresh also keeps you on final state ... I wanted to be able to go back to initial page + combined with the logo reloading the page without "smart" consolidation of the state will create a more rich browser history so people can return to each restart in their journey (if someone wants to do multiple restarts by clicking on the logo - probably very niche use case)

so not polluting the browser history with every state change while remembering both the initial and the final state sounded like a good compromise to me 🤞